### PR TITLE
implement helper functions for tap detection

### PR DESCRIPTION
**Pull request by dtimm**

### CODE REVIEW

**Review by sajattack — APPROVED** [review]

**Comment by sajattack** [comment]
LGTM but I'll see if @BenBergman wants to review before merging.

**Comment by BenBergman** [comment]
At this point, @sajattack is the defacto maintainer of this repository. I haven't touched it pretty much since I created it.

Feel free to merge anything you want, with or without my input.

**Pull request merged**
